votepy/generic_ilp.py

```diff
@@ -0,0 +1,141 @@
+from abc import ABC,abstractmethod
+from enum import Enum
+from typing import Optional
+
+class Solver(ABC):
+    class Sense(Enum):
+        MAX = 1
+        MIN = 2
+    @abstractmethod
+    def addVariable(self, name: str, vartype: str, obj: Optional[float], lb: Optional[float], ub: Optional[float]):
```

Comment:
Yeah, these are taken directly from the CPLEX API and they represent lower and upper bound of a variable. But that probably means that some docstrings should be added there. I skipped them for now since it's a class intended for internal use only as of right now